internal/testcomponents/example_factories.go

```diff
@@ -39,3 +48,52 @@ func ExampleComponents() (
 
 	return
 }
+
+// DefaultComponents returns the default set of components used by the OpenTelemetry collector.
+func DefaultComponents() (
```

Comment:
probably this needs to be in a different package than the example/testing components.

Reply:
Does this need to live elsewhere for `cmd/otelcol` to access it?  That is going away.  If it needs to live elsewhere for other things to access then I think we need a shared understanding of what those things are and how they will utilize it.

Reply:
@bogdandrutu This has been moved to internal package. Thank you.